src/modules/SX126x/SX126x.cpp

```diff
@@ -598,6 +608,14 @@ int16_t SX126x::startReceiveCommon() {
   // clear interrupt flags
   state = clearIrqStatus();
 
+  // set implicit mode and expected len if applicable
+  if (_headerType == SX126X_LORA_HEADER_IMPLICIT && getPacketType() == SX126X_PACKET_TYPE_LORA) {
```

Comment:
`getPacketType` is an extra SPI expense but I think it's necessary. Is it?

Reply:
You're right, it is necessary in this case.